### PR TITLE
Added notifier for HipChat

### PR DESCRIPTION
**Pull request by ukjbrown**
I've added notification support for Atlassian HipChat. I'm using this in a live project, so thought I'd share it if you're interested.

### CODE REVIEW

File: spring-boot-admin-server/src/test/java/de/codecentric/boot/admin/notify/HipchatNotifierTest.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        expected.put("color", "green");
+        expected.put("message", "<strong>App</strong>/-id- is <strong>UP</strong>");
+        expected.put("notify", Boolean.TRUE);
+        expected.put("message_format", "html");
+
+        verify(restTemplate).postForEntity(any(String.class), eq(expected), eq(Void.class));
+    }
+
+    @Test
+    public void test_onApplicationEvent_trigger()
```
Comment (joshiste) [inline spring-boot-admin-server/src/test/java/de/codecentric/boot/admin/notify/HipchatNotifierTest.java]: Codacy Issue found: [Method names should not contain underscores](https://www.codacy.com/app/joshiste/spring-boot-admin/file/2237865097/issues/source?bid=3175464&fileBranchId=3178546#l59)

File: spring-boot-admin-server/src/test/java/de/codecentric/boot/admin/notify/HipchatNotifierTest.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        notifier = new HipchatNotifier();
+        notifier.setNotify(true);
+        notifier.setAuthToken("--token-");
+        notifier.setRoomId("-room-");
+        notifier.setUrl(URI.create("http://localhost/v2"));
+        notifier.setRestTemplate(restTemplate);
+    }
+
+    @Test
+    public void test_onApplicationEvent_resolve()
```
Comment (joshiste) [inline spring-boot-admin-server/src/test/java/de/codecentric/boot/admin/notify/HipchatNotifierTest.java]: Codacy Issue found: [Method names should not contain underscores](https://www.codacy.com/app/joshiste/spring-boot-admin/file/2237865097/issues/source?bid=3175464&fileBranchId=3178546#l40)

Comment (ukjbrown) [inline spring-boot-admin-server/src/test/java/de/codecentric/boot/admin/notify/HipchatNotifierTest.java]: Not sure if these comments are automatic, but the underscore convention was copied from the other Notifier Test classes. I wouldn't ordinarily use them myself unless implementing Cucumber tests, but figured this style was preferred.

**Comment by joshiste** [comment]
Hmm the codacy integration is new. Maybe some of the rules need to be tweaked...

**Comment by joshiste** [comment]
Awsome! Thanks for sharing!

**Pull request merged**